design/one-pager-easy-stacks.md

```diff
@@ -0,0 +1,294 @@
+abilites, to deploy an application and all other things that you can do with
+Kubernetes controllers since main components of stacks are CRDs and controllers
+watching them.
+
+To achieve the mentioned goals, we can publish the YAML files in a stack with a
+controller that has its own CRD with high level parameters. Controller deploys
+and updates the YAMLs that are packaged in the stack image according to user's
+input on the CR instance of that stack that user creates.
+
+As sky is the limit when it comes to stacks, we need to put down some general
+rules to be adhered by initial configuration stacks.
+
+* CRDs of the resources that will be deployed by the stack are declared as
+  dependency so that user knows what CR instances will be deployed by the given
+  configuration stack.
+* It should be possible to have different sets of configurations in one stack
+  and user should be able to choose from them via the CR that is an instance
+  of stack's CRD.
+  * Note that it's highly preferred to have 1 custom resource definition that
+    the stack uses and expose the high level variations of the parameters through
+    `spec` instead of having different custom resource definitions for each class
+    of configuration.
+* A minimal high level set of configurations are exposed to the user through
+  the stack's CR, such as `Provider` reference to refer in the deployed resource
+  or `region`.
+* Controller should signal readiness of all resources in CR's status.
+* All resources should be labelled referring to the CR instance of the stack.
+* In YAMLs, resources should refer to each other using cross-resource references
+  wherever possible.
+* **TBD** Controller should update the resources continuously and treat the given CR as
+  the source of truth even though user manually changes the resources that are
+  deployed by the stack.
+* **TBD** CRD of the configuration stack should be namespaced so that different
+  teams/users can create their own configured environment.
+* Stack type will be `ClusterStack` as most of the resources it deploys are
+  cluster-scoped.
+
+Name of the concept is **TBD**; _EasyStack_, _ConfigurationStack_,
+_EnvironmentStack_ are current candidates.
+
+## User Experience
+
+1. Install AWS stack and create your `Provider` object named `aws-provider`.
+2. Install a AWS configuration stack via:
+```yaml
+# exact apiVersion is TBD.
+apiVersion: stacks.crossplane.io/v1alpha1
+kind: ClusterStackInstall
+metadata:
+  name: easy-aws
+  namespace: crossplane-system
+spec:
+  package: "crossplane/easy-aws:latest"
+```
+
+Now, I want an environment where all my database instances and kubernetes clusters
+are connected to each other in a private VPC, which what this specific configuration
+stack does. Create the following:
+
+```yaml
+# exact apiVersion is TBD.
+apiVersion: aws.configurationstacks.crossplane.io/v1alpha1
+kind: EasyAWS
+metadata:
+  name: project-future
+  namespace: dev
```

Comment:
per comment above -- an environment (dev/test, stage, prod) generally maps to crossplane instance, so could this be cluster-scoped? If not perhaps we should recommend it go into the crossplane-system namespace [like we do for provider secret today](https://github.com/crossplaneio/crossplane/blob/master/docs/cloud-providers/aws/aws-provider.md#optional-setup-aws-provider-manually).